flybywiresim/a32nx
# feat(fadec): Impl FADEC GND PWR buttons + corrected Igniter logic

Fixes #[issue_no]

## Summary of Changes
This PR implements the FADEC GND PWR pushbuttons for each FADEC. It also implements the autoswitching logic 
of igniters on an automatic groundstart. Previously it would always display both igniters in use on startup.
Now it only uses one igniter and alternates after every startup attempt.

Crankmode now also energizes the FADEC.

I also moved the igniter code into the A32NX_FADEC.js file since it seemed more appropriate. 

## Screenshots (if necessary)


## References
Since I didn't have access to NEO documentation, most of the logic here is based on videos and publicly available CEO documentation. Should any discrepancy be discovered I'd be happy to adjust my code.




## Additional context



Discord username (if different from GitHub):

## Testing instructions


The test regiment should manly deal with FADEC related logic.

1. Spawn C/D
2. Power Up A/C using ground power
3. Wait 5 minutes for initial FADEC power up period to elapse and EICAS to show amber X
4. Use the FADEC GND PWR 1+2 Pushbuttons to activate the respective Fadecs and watch the EICAS show values
5. Turn off pushbuttons and see amber X on EICAS
6. Move Engine Mode Selector into CRANK 
7. Watch EICAS display values instead of amber X
8. Move Engine Mode Selector to NORM 
9. Watch EICAS display amber X
10. Move Engine Mode Selector to IGN/START
11. Watch EICAS display values instead of amber X
12. Move Engine Mode Selector to NORM 
13. Wait 30 seconds
14. Watch EICAS display amber X
15. Turn on APU and APU BLEED and turn on Fuel Pumps
16. Move Engine Mode Selector to IGN/START
17. Move Engine Master SW to ON
18. Watch EICAS until N2 reaches > 18%
19. Watch bottom of EICAS for either A or B to be displayed next to the Valve state
20. Watch EICAS until N2 reaches > 55%
21. Watch bottom of EICAS for either A or B to disappear
22. Once Engine is started, turn Engine Mode Selector to NORM
23. Shut it down by turning off the corresponding Engine Master SW
24. Engine should shutdown and 5 minutes after the Engine Master SW was turned off the FADEC should unpower and the EICAS should show amber X
25. Repeat steps 16-18
26. This time the opposite Igniter should be used i.e B when it was A or vice versa 
27. Repeat for Engine 2


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
